### PR TITLE
BAEL-8021-Guide-to-CompletableFuture-join-vs-get

### PR DESCRIPTION
**Pull request by vunamtien**

### CODE REVIEW

File: core-java-modules/core-java-concurrency-advanced-5/src/test/java/com/baeldung/joinvsget/CompletableFutureUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        try {
+            assertEquals("Test get", future.get());
+        } catch (InterruptedException | ExecutionException e) {
+            fail("Exception should not be thrown");
+        }
+
+        CompletableFuture<String> exceptionFuture = CompletableFuture.supplyAsync(() -> {
+            throw new RuntimeException("Test get exception");
+        });
```
Comment (pivovarit) [inline core-java-modules/core-java-concurrency-advanced-5/src/test/java/com/baeldung/joinvsget/CompletableFutureUnitTest.java:36]: Try ```CompletableFuture.failedFuture(new RuntimeException("Test get exception"))```

File: core-java-modules/core-java-concurrency-advanced-5/src/test/java/com/baeldung/joinvsget/CompletableFutureUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Test
+    public void givenJoinMethod_whenThrow_thenGetUncheckedException() {
+        CompletableFuture<String> future = CompletableFuture.supplyAsync(() -> "Test join");
+        
+        assertEquals("Test join", future.join());
+
+        CompletableFuture<String> exceptionFuture = CompletableFuture.supplyAsync(() -> {
+            throw new RuntimeException("Test join exception");
+        });
```
Comment (pivovarit) [inline core-java-modules/core-java-concurrency-advanced-5/src/test/java/com/baeldung/joinvsget/CompletableFutureUnitTest.java:19]: Try ```CompletableFuture.failedFuture(new RuntimeException("Test join exception"))```

**Comment by lor6** [comment]
retest this please

**Comment by pivovarit** [comment]
retest this please

**Pull request merged**